src/coreclr/debug/di/rsthread.cpp

```diff
@@ -633,7 +633,7 @@ void CordbThread::RefreshHandle(HANDLE * phThread)
     HANDLE hThread = pDAC->GetThreadHandle(m_vmThreadToken);
 
     _ASSERTE(hThread != INVALID_HANDLE_VALUE);
-    PREFAST_ASSUME(hThread != NULL);
+    COMPILER_ASSUME(hThread != NULL);
```

Comment:
`COMPILER_ASSUME` is meant to be used only in a micro-optimized code where you want to give compiler hints about invariants guaranteed by the code so that the compiler can perform additional optimizations.

I do not see any of the new `COMPILER_ASSUME`s to be like that. I would replace them with regular `_ASSERTE`s so that people are not puzzled about the purpose of `COMPILER_ASSUME`.